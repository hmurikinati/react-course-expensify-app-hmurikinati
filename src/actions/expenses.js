import uuid from 'react-uuid';
import database from '../firebase/firebase'

export const addExpense = (expense) => ({
    type: 'ADD_EXPENSE',
    expense
});

export const startAddingExpense = (expenseData = {}) => {
    console.log('expenseData :', expenseData);
    return (dispatch) =>{
        const {
            description = '',
            notes = '',
            amount = 0,
            createdAt = 0
        } = expenseData;
        const expense = {description, notes, amount, createdAt};
        return database.ref('expenses').push(expense)
        .then((ref) => {
            dispatch(addExpense({
                id : ref.key,
                ...expense
            }));
        });      
      };
};

export const removeExpense = ({ id = 'aaa' } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});

export const editExpense = ({ id }, updates) => ({
    type: "EDIT_EXPENSE",
    id,
    updates
});