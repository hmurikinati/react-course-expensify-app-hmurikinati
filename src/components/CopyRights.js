import React from 'react';
import Link from '@material-ui/core/Link';
import Typography from '@material-ui/core/Typography';

const CopyRights = () => {
    return (
      <Typography variant="body2" color="textSecondary" align="center">
        {'Copyright Â© '}
        <Link color="inherit" href="https://material-ui.com/">
          Expensify
        </Link>{' '}
        {new Date().getFullYear()}
        {'.'}
      </Typography>
    );
  }
  export default CopyRights;