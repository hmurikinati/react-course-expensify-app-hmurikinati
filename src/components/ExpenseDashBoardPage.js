import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilters';
const ExpenseDashBoardPage = () => (
    <div>
        <ExpenseListFilter />
        <ExpenseList />
    </div>
);

export default ExpenseDashBoardPage;