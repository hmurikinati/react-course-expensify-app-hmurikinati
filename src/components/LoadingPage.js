import React from 'react';

export const LoadingPage = () => {
    return (
        <div className="loader">
           <img className="loader__image" src="/images/loading.gif"></img>
        </div>
    )
}

export default LoadingPage;