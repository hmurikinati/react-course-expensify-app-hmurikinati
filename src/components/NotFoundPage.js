import React from 'react';
import {Link } from 'react-router-dom';


const NotFoundPage = () => (
    <div>
        No Match Found <Link to='/'>Go Home</Link>
    </div>
);

export default NotFoundPage;