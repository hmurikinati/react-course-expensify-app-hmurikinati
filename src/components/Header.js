import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header>
        <header>
            <h1>Expensify</h1>
        </header>
        <NavLink to='/dashboard' activeClassName='is-active'>Dashboard</NavLink>
        <NavLink to='/create' activeClassName='is-active'>Create Expense</NavLink>
        <NavLink to='/help' activeClassName='is-active'>Help</NavLink>
        <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});
export default connect(undefined, mapDispatchToProps)(Header);