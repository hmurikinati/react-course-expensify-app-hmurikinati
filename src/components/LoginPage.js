import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => {
   
    return (
        <div>
            <button className='btn btn-primary' onClick={startLogin}>Login</button>
        </div>
    );
}
const mapDispatchToProps = (dispatch) => ({ 
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);