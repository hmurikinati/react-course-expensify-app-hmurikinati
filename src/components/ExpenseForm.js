import React from 'react';
import moment from 'moment';
import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';
import { Header } from './Header';


import 'react-dates/lib/css/_datepicker.css';
const now = moment();


export default class ExpenseForm extends React.Component {

    constructor(props) {

        super(props);
        this.state = {
            description: (props.expense && props.expense.description) ? props.expense.description : '',
            notes: (props.expense  && props.expense.notes) ? props.expense.notes : '',
            amount: (props.expense  && props.expense.amount) ? (props.expense.amount / 100).toString() : '',
            createdAt: ( props.expense && props.expense.createdAt )? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        }

    }


    descriptionOnchange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));

    }

    noteOnChange = (e) => {
        const notes = e.target.value;
        this.setState(() => ({ notes }));
    }
    onAmountChange = (e) => {
        const amount = e.target.value;

        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));
        }
    }
    onDateChange = (createdAt) => {
        
        if (createdAt) {
            this.setState(() => ({ createdAt }));
        }
    }
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }))

    }
    onSubmit = (e) => {
        e.preventDefault();
       
        if (!this.state.description || !this.state.amount) {
            this.setState(() => ({ error: 'Please enter Description and AMount' }));
        } else {
            this.setState(() => ({ error: '' }));
           
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                notes: this.state.notes,
                createdAt: this.state.createdAt.valueOf()
            })
        }
    }
    render() {
        return (
            <div>
            {/* <Header /> */}
                {this.state.error && <p> {this.state.error} </p>}
                <form onSubmit={this.onSubmit}>
                    <input
                        type='text'
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.descriptionOnchange}
                    />

                    <input
                        type='text'
                        placeholder='Amount'
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                    <textarea
                        placeholder='Add a note for your expense (optional)'
                        value={this.state.notes}
                        onChange={this.noteOnChange}
                    />
                    <button>Save Expense</button>
                    
                </form>
            </div>
        )
    }
}