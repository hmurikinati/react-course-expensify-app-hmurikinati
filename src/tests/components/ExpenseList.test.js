import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses'

test('Should render Expense list  correctly', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />)
    expect(wrapper).toMatchSnapshot()
});



test("should render expense list with empty message", () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />)
    expect(wrapper).toMatchSnapshot()
});