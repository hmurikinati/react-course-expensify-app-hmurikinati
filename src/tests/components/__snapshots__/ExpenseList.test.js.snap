// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expense list  correctly 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <ExpenseListItem
    amount={2000}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    notes=""
  />
  <ExpenseListItem
    amount={3000}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    notes=""
  />
  <ExpenseListItem
    amount={1000}
    createdAt={345600000}
    description="Credit card"
    id="3"
    key="3"
    notes=""
  />
</div>
`;

exports[`should render expense list with empty message 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <p>
    No Expense found..
  </p>
</div>
`;
